metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala

```diff
@@ -71,6 +76,17 @@ final class OrganizeImports(
 
   }
 
+  private def hasUnusedImport(
+      params: CodeActionParams
+  ): Boolean = {
+    params
+      .getContext()
+      .getDiagnostics()
+      .asScala
+      .filter(_.getMessage().toLowerCase == OrganizeImports.unusedImport)
```

Comment:
This looks ok, but do you think it may make sense to use the same unapply pattern we use for other scalac diagnostics?

Reply:
Yea I went back and forth. For a case like this where we know exactly what it will be, is there a benefit using the unapply for this?

Reply:
Mostly consistency, plus the fact that is tucked away and we can complicate the check if needed (e.g. in case it requires more than a substring check)

Reply:
Sounds good. Done and Done.